packages/gatsby-theme-api-docs/src/utils/extract-additional-info.spec.js

```diff
@@ -0,0 +1,66 @@
+import extractAdditionalInfo from './extract-additional-info';
+
+describe('Ordering tags', () => {
+  const data = createTestData();
+
+  it('should order default property to the very top', () => {
+    const sortedData = extractAdditionalInfo(data[0]);
+    expect(JSON.stringify(sortedData)).toBe(
+      JSON.stringify(expectedTestResults()[0])
+    );
+  });
+  it('should order properties with min and max names to the top behind the default', () => {
+    const sortedData = extractAdditionalInfo(data[1]);
+    expect(JSON.stringify(sortedData)).toBe(
+      JSON.stringify(expectedTestResults()[1])
+    );
+  });
+  it('should order properties with min and max names to the very top if there is no default property', () => {
+    const sortedData = extractAdditionalInfo(data[2]);
+    expect(JSON.stringify(sortedData)).toBe(
+      JSON.stringify(expectedTestResults()[2])
+    );
+  });
+});
+
+function createTestData() {
+  return [
+    {
+      maxItems: 1,
+      minItems: 2,
+      uniqueItems: 2,
+      default: 1,
+    },
+    {
+      maximum: 1,
+      default: 1,
+      minimum: 2,
+    },
+    {
+      minLength: 1,
+      pattern: 2,
+      maxLength: 1,
+    },
+  ];
+}
+
+function expectedTestResults() {
+  return [
+    {
```

Comment:
you're comparing object keys here.  An object is identical no matter how it's ordered.  Javascript objects are not ordered.  You need to migrate the data structure to an array (see my previous review). 
Stringifying in the middle makes it even less predictable.  

Example array structure:
```js
const currentlyNoOrderInformation = {
  maxItems: 1,
  minItems: 2
}  // if you now change the order, it's still the same object

const orderedInformationExample = [
  { name: "maxItems", value: 1},
  { name: "minItems", value: 2}
]  // if you now change the order, it's not the same array any more

```

JSON spec (since you're serializing and deserializing here): https://www.rfc-editor.org/rfc/rfc7159.html#section-1  
> An object is an unordered collection of zero or more name/value
   pairs, where a name is a string and a value is a string, number,
   boolean, null, object, or array.

Reply:
@timonrey  PS for more exact precision:
 * since ES 2015 _string_ keys in objects are retained in their _insertion_ order.   
 * JSON never allowed assumptions about object key ordering

In practice even JSON keeps the order in most implementations but again, it's a bit of a guess. 

What I wanted to primarily say: It's a bit of guesswork to find the root cause if you are not 100% sure the language features are used correctly.  You could start by removing the back and forth via JSON but then you would still have the unclear situation whether the `equals()` deep comparison is checking object order or not.  I think it shouldn't.